biblib/views/document_view.py

```diff
@@ -44,20 +45,174 @@ def add_document_to_library(cls, library_id, document_data):
 
             start_length = len(library.bibcode)
 
-            library.add_bibcodes(document_data['bibcode'])
+            #Validate supplied bibcodes to confirm they exist in SOLR
+            solr_resp, status_code = cls.query_valid_bibcodes(document_data['bibcode'])
 
-            session.add(library)
-            session.commit()
+            if "error" in solr_resp.keys():
+                #If SOLR request fails, pass the error back to the user
+                current_app.logger.error("Failed to retrieve bibcodes with error: {}".format(solr_resp.get("error")))
+                output_dict = {"error": solr_resp.get("error"), "number_added": 0, "status": status_code}
+                valid_bibcodes = []
+                return output_dict
+            else:
+                #If SOLR query succeeds generate list of valid bibcodes from response
+                valid_bibcodes = [doc.get('bibcode') for doc in solr_resp.get('docs', {})]
+                current_app.logger.info("Found the following valid bibcodes: {}".format(valid_bibcodes))
+            
+            if valid_bibcodes:
+                #Add all valid bibcodes to library
+                library.add_bibcodes(valid_bibcodes)
+
+                session.add(library)
+                session.commit()
+
+                current_app.logger.info('Added: {0} is now {1}'.format(
+                    valid_bibcodes,
+                    library.bibcode)
```

Comment:
`library.bibcode` can be giant for big libraries - I'd move this part of the log statement to a separate statement at the `logger.debug` level instead. The first part of the log statement (`valid_bibcodes`) can/should stay at the `logger.info` level though